### PR TITLE
Recreate missing java.io.tmpdir when creating LDD download temp files

### LINKED ISSUE #142: Harvest does not create LDD download temp files on Windows, causing all products in a bundle to fail
## Checked for duplicates
Yes - I've already checked

## 🐛 Describe the bug
When Harvest needs to download an LDD (e.g. because it isn't yet loaded in the target registry), it calls `SchemaUpdater.createLddTempFile()` (`common/src/main/java/gov/nasa/pds/registry/common/es/service/SchemaUpdater.java`). That method first tries `Files.createTempFile(..., PosixFilePermissions...)`, and on `UnsupportedOperationException`/`IOException` falls back to plain `File.createTempFile(...)`. **Both calls rely on the same JVM default temp directory (`java.io.tmpdir`) with no override.**

On at least one Windows harvest host, both attempts fail with:
```
IOException: The system cannot find the path specified
```
so `createLddTempFile` throws an `LddException` for every product that triggers an LDD update. Because no previously-loaded LDD exists for the namespace and force-load mode is off, this escalates to "No previously loaded LDD found for namespace '\<prefix\>'. Cannot load products with fields from this namespace," which propagates up and is caught in `ProductProcessor.onFile()`, logged as an ERROR, and counted as a failed file — for every single product in the bundle.

Net effect: the entire harvest run fails (0 products loaded) even though the actual referenced LDD schema itself is valid and reachable — the real problem (an unwritable/nonexistent default temp directory on the host) is masked by a generic OS error with no indication of which directory was attempted or how to work around it.

This appears to be a regression introduced by the recent `registry-common#304` port that added the POSIX-permission-then-fallback logic in `createLddTempFile()` (commits `52b43786` and `9cae169d` in this repo) — the fallback path doesn't actually provide a working alternative when the default temp directory itself is the problem.

## 🕵️ Expected behavior
- Harvest should not fail 100% of products in a bundle because of a single temp-directory issue.
- The error message should identify the actual temp directory path that was attempted, so operators can diagnose/fix the underlying environment problem.
- Ideally, the temp directory used for LDD downloads should be configurable (e.g. via Harvest config or a system property), rather than being hardcoded to whatever `java.io.tmpdir` resolves to.

## 📜 To Reproduce
1. On a Windows host where the resolved `java.io.tmpdir` path does not exist or is not writable, run Harvest against a bundle that requires updating/downloading an LDD not already loaded in the target registry.
2. Observe repeated ERROR log lines such as:
   ```
   [ERROR] (ProductProcessor.java:onFile:173) Failed to create temp file for LDD download for namespace '<namespace>': The system cannot find the path specified
   ```
3. Every product in the bundle fails to load. In the reported run: Skipped files: 7, Loaded files: 0, Failed files: 7834.

## 🖥 Environment Info
- Version of this software: Harvest 5.1.3
- Operating System: Windows Server (harvest installed under a local drive, data referenced from a mapped network share)
- Registry backend: OpenSearch

## 📚 Version of Software Used
Harvest 5.1.3

## 🩺 Test Data / Additional context
- Reported via email by a PDS Geosciences Node operator while harvesting bundle `urn:nasa:pds:mer1_mtes_derived_emissivity`, referencing LDD `msn_surface` v1 (`PDS4_MSN_SURFACE_1A10.xsd`).
- Full harvest log (~31k lines, all repeating the same temp-file error) is available on request; a sanitized excerpt:
  ```
  2026-08-07 13:00:50 [INFO ] (SchemaUpdater.java:updateSchema:95) Updating LDDs.
  2026-08-07 13:00:50 [INFO ] (SchemaUpdater.java:updateLdd:166) Updating 'msn_surface' LDD. Schema location: https://pds.nasa.gov/pds4/mission/msn_surface/v1/PDS4_MSN_SURFACE_1A10.xsd
  2026-08-07 13:00:51 [ERROR] (ProductProcessor.java:onFile:173) Failed to create temp file for LDD download for namespace 'msn_surface': The system cannot find the path specified
  ```
- Relevant code: `common/src/main/java/gov/nasa/pds/registry/common/es/service/SchemaUpdater.java:332-352` (`createLddTempFile`).

## 🦄 Related requirements
N/A

---

### PR DESCRIPTION
**Pull request by jordanpadams**
## 🗒️ Summary
`SchemaUpdater.createLddTempFile()` downloads an LDD to a temp file, trying `Files.createTempFile()` with POSIX permissions first and falling back to plain `File.createTempFile()` on failure. Both calls use the JVM's default `java.io.tmpdir` with no directory override, so on a host where that directory doesn't exist, both attempts fail identically with `IOException: The system cannot find the path specified` — and every single product that needs an LDD update fails.

This is exactly what was reported in #142: a PDS Geosciences Node operator running Harvest 5.1.3 on Windows saw a bundle with 7,834 products fail 100% (0 loaded) because of this one systemic temp-directory issue, even though the referenced `msn_surface` LDD schema itself was valid and reachable.

Fix: before attempting temp file creation, ensure the resolved `java.io.tmpdir` directory exists (`mkdirs()`), and include the resolved path in the `LddException` message if creation still fails, so the failure is actionable. This is a straight port of the same fix in NASA-PDS/registry-common#305 (reformatted to this file's brace/indent style) — `registry-loader`'s `common` module carries a ported copy of `SchemaUpdater`, and `harvest` does not need any change since it consumes `registry-common` as a Maven dependency and contains no copy of this code.

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [x] AI generated substantial portions of this code

Estimated % of code influenced by AI: 90 %

## ⚙️ Test Data and/or Report
Added `createLddTempFile_recreatesMissingTmpDir` to `common/src/test/java/service/TestSchemaUpdaterTempFile.java`, which points `java.io.tmpdir` at a directory that does not yet exist and asserts `createLddTempFile()` creates it and succeeds. Ran locally:

```
mvn test -pl common -am -Dtest=TestSchemaUpdaterTempFile
...
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0
BUILD SUCCESS
```

## ♻️ Related Issues
Fixes #142

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: common/src/main/java/gov/nasa/pds/registry/common/es/service/SchemaUpdater.java [hunk 1]
```diff
@@ -331,6 +331,13 @@ private String getJsonUrl(String uri)
      */
     public static File createLddTempFile(String prefix) throws LddException
     {
+        String tmpDirPath = System.getProperty("java.io.tmpdir");
+        File tmpDir = new File(tmpDirPath);
+        if(!tmpDir.exists())
+        {
+            tmpDir.mkdirs();
```
Comment (nutjob4life) [inline common/src/main/java/gov/nasa/pds/registry/common/es/service/SchemaUpdater.java:338]: Déjà vu: this returns a boolean and `false` can indicate failure, but this is _so highly unlikely_ given that `java.io.tmpir` is almost always a writeable location. I wouldn't bother, but I am a pedant as you know 🤭

**Review by nutjob4life — APPROVED** [review]
Wondering if this should be refactored into a registry-common-common: nah
Déjà vu: experienced
Pedant comment: added
Code delta: confirmed
Test case: actually quite good
Maven tests: passing
Overall approval: ✅
Maven details:
```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for gov.nasa.pds:registry-loader 1.4.0-SNAPSHOT:
[INFO] 
[INFO] gov.nasa.pds:registry-loader ....................... SUCCESS [  0.233 s]
[INFO] PDS Registry Common ................................ SUCCESS [ 32.381 s]
[INFO] Harvest Tool ....................................... SUCCESS [  3.303 s]
[INFO] PDS Registry Manager ............................... SUCCESS [  2.774 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
```
